File: frontend/src/modules/cruise-applications/models/PermissionDto.ts
import { z } from 'zod';

import { FileDto, FileDtoValidationSchema } from '@/cruise-applications/models/FileDto';

export type PermissionDto = {
  description: string;
  executive: string;
  scan?: FileDto | undefined;
};

export const PermissionDtoValidationSchema = z.object({
  description: z.string().nonempty('Treść pozwolenia jest wymagany').max(10240, 'Maksymalna długość to 10240 znaków'),
  executive: z.string().nonempty('Organ wydający jest wymagany'),
  scan: FileDtoValidationSchema.optional(),
});

export const PermissionDtoWithFileValidationSchema = PermissionDtoValidationSchema.extend({
  scan: FileDtoValidationSchema.extend({
    name: z.string().endsWith('.pdf', 'Plik musi być w formacie PDF'),
  }),
});

File: backend/ResearchCruiseApp/Api/Applications/Contracts/PermissionDto.cs
using System.ComponentModel.DataAnnotations;

namespace ResearchCruiseApp.Api.Applications.Contracts;

public class PermissionDto
{
    [StringLength(1024)]
    public string? Description { get; init; }

    [StringLength(1024)]
    public string? Executive { get; init; }

    public FileDto? Scan { get; set; }
}
